### PR TITLE
Initial version of the service

### PR DESCRIPTION
**Pull request by nwrigh**
This service is to write the most TME types to Neo4j

### CODE REVIEW

File: concepts/concepts_service.go [hunk 1]
```diff
@@ -245,17 +245,25 @@ func getAllLabels(conceptType string) string {
 }
 
 func createNodeQueries(concept Concept) []*neoism.CypherQuery {
+	var labelsToRemove string
+	for i, conceptType := range conceptLabels {
```
Comment (david-smith-ft) [inline concepts/concepts_service.go]: You can do this with ` labelsToRemove := ":" + strings.Join(conceptLabels, ":")`

File: main.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	log.SetLevel(log.InfoLevel)
+	log.Infof("Application started with args %s", os.Args)
+	app.Run(os.Args)
+}
+
+func makeCheck(service baseftrwapp.Service, cr neoutils.CypherRunner) v1a.Check {
+	return v1a.Check{
+		BusinessImpact:   "Cannot read/write concepts via this writer",
+		Name:             "Check connectivity to Neo4j - neoUrl is a parameter in hieradata for this service",
+		PanicGuide:       "TODO - write panic guide",
```
Comment (tosan88) [inline main.go]: Could you please create a basic runbook for the service and link that here?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+FROM golang:1.8-alpine
```
Comment (tosan88) [inline Dockerfile]: Seems that the project is set up for vendoring, but it isn't used here. Could you please changed that? For example see : https://github.com/Financial-Times/post-publication-combiner/blob/master/Dockerfile? (didn't find any documentation around it for us)

File: README.md [hunk 4]
```diff
@@ -1 +1,93 @@
-# concepts-rw-neo4j
+# Concept Reader/Writer for Neo4j (concept-rw-neo4j)
+
+[![Circle CI](https://circleci.com/gh/Financial-Times/concept-rw-neo4j.svg?style=shield)](https://circleci.com/gh/Financial-Times/concept-rw-neo4j)[![Go Report Card](https://goreportcard.com/badge/github.com/Financial-Times/people-rw-neo4j)](https://goreportcard.com/report/github.com/Financial-Times/people-rw-neo4j) [![Coverage Status](https://coveralls.io/repos/github/Financial-Times/people-rw-neo4j/badge.svg)](https://coveralls.io/github/Financial-Times/people-rw-neo4j)
```
Comment (tosan88) [inline README.md]: There are some references to `people-rw-neo4j`

File: README.md [hunk 5]
```diff
@@ -1 +1,93 @@
...
+__An API for reading/writing concepts into Neo4j. 
+
+## Installation
+
+For the first time:
+
+`go get github.com/Financial-Times/concepts-rw-neo4j`
```
Comment (tosan88) [inline README.md]: You might want to add steps for go vendoring here as well

File: main.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+		Desc:   "Maximum number of statements to execute per batch",
+		EnvVar: "BATCH_SIZE",
+	})
+	logMetrics := app.Bool(cli.BoolOpt{
+		Name:   "logMetrics",
+		Value:  false,
+		Desc:   "Whether to log metrics. Set to true if running locally and you want metrics output",
+		EnvVar: "LOG_METRICS",
+	})
+	env := app.String(cli.StringOpt{
```
Comment (tosan88) [inline main.go]: I thought *env* was used only in FTP2 world. 
Will this be ever something else than "local" in our stack? If no, I would inline it & remove this variable

File: main.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	_ "net/http/pprof"
+	"os"
+
+	"github.com/Financial-Times/base-ft-rw-app-go/baseftrwapp"
+	"github.com/Financial-Times/concepts-rw-neo4j/concepts"
+	"github.com/Financial-Times/go-fthealth/v1a"
```
Comment (tosan88) [inline main.go]: Could you please use the latest version (`the github.com/Financial-Times/go-fthealth/v1_1`)? https://github.com/Financial-Times/go-fthealth/blob/master/README.md

File: concepts/model.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+	SourceRepresentations []Concept `json:"sourceRepresentations"`
+}
+
+// Concept - could be any concept genre, subject etc
+type Concept struct {
+	UUID           string `json:"uuid"`
+	PrefLabel      string `json:"prefLabel"`
+	Type           string `json:"type"`
+	Authority      string `json:"authority"`
+	AuthorityValue string `json:"authorityValue"`
```
Comment (tosan88) [inline concepts/model.go]: Is this the same as `identifierValue`? If yes, shouldn't we be consistent about the naming (I don't have a strong opinion on that though) ?

File: circle.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  pre:
+    - go get github.com/axw/gocov/gocov; go get github.com/matm/gocov-html; go get -u github.com/jstemmer/go-junit-report
+test:
+  pre:
+    - docker info; docker run -d -e "NEO4J_AUTH=none" -e "NEO4J_HEAP_MEMORY=256" -e "NEO4J_CACHE_MEMORY=256M" -p 7474:7474 -p 7473:7473 neo4j:$NEO4J_VERSION
+    - wget --retry-connrefused --no-check-certificate -T 60 $NEO4J_TEST_URL; curl $NEO4J_TEST_URL
+    - go get github.com/mattn/goveralls
+  override:
+    - mkdir -p $CIRCLE_TEST_REPORTS/golang
+    - go test -race -v ./...
```
Comment (tosan88) [inline circle.yml]: Go vendoring here as well...

**Pull request merged**